src/runtime-rs/crates/hypervisor/src/device/driver/vfio.rs

```diff
@@ -22,56 +73,533 @@ fn override_driver(bdf: &str, driver: &str) -> Result<()> {
     Ok(())
 }
 
-const SYS_PCI_DEVICES_PATH: &str = "/sys/bus/pci/devices";
-const PCI_DRIVER_PROBE: &str = "/sys/bus/pci/drivers_probe";
-const VFIO_NEW_ID_PATH: &str = "/sys/bus/pci/drivers/vfio-pci/new_id";
-const VFIO_UNBIND_PATH: &str = "/sys/bus/pci/drivers/vfio-pci/unbind";
-
-pub const VFIO_PCI: &str = "vfio-pci";
-
-#[derive(Debug, Clone)]
+#[derive(Clone, Debug, Default, PartialEq)]
 pub enum VfioBusMode {
-    PCI,
+    #[default]
     MMIO,
+    PCI,
 }
 
 impl VfioBusMode {
-    pub fn new(mode: &str) -> Result<Self> {
-        Ok(match mode {
-            "mmio" => VfioBusMode::MMIO,
-            _ => VfioBusMode::PCI,
-        })
+    pub fn new(mode: &str) -> Self {
+        match mode {
+            "c" | "pci" => VfioBusMode::PCI,
+            _ => VfioBusMode::MMIO,
+        }
+    }
+
+    pub fn revert(mode: VfioBusMode) -> String {
+        match mode {
+            VfioBusMode::PCI => "pci".to_owned(),
+            VfioBusMode::MMIO => "mmio".to_owned(),
+        }
+    }
+
+    pub fn driver_type(mode: &str) -> &str {
+        match mode {
+            "c" | "pci" => DRIVER_VFIO_PCI_TYPE,
+            _ => DRIVER_MMIO_BLK_TYPE,
```

Comment:
I think this is for `virito-blk`, should we remove this?

Reply:
Thx, 
I'm sorry that it will be reserved, because it's for vfio volume when kata-agent handle it with add_storage/mmioblk.